src/screens/Assets/AssetSearch.js

```diff
@@ -0,0 +1,408 @@
+  align-items: center;
+`;
+
+const SearchSpinner = styled(Wrapper)`
+  padding-top: 20;
+`;
+
+class AssetSearch extends React.Component<Props, State> {
+  displayedAssets: Asset[];
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeTab: DISPLAYED,
+      query: '',
+    };
+
+    const { supportedAssets, assets } = props;
+    const displayedAssetsSymbols = Object.keys(assets);
+    this.displayedAssets = displayedAssetsSymbols.map(symbol => supportedAssets.find(asset => asset.symbol === symbol));
+    this.doAssetsSearch = debounce(this.doAssetsSearch, 500);
+  }
+
+  componentDidUpdate() {
+    const { assets } = this.props;
+
+    Object.keys(assets).forEach(symbol => {
+      if (!this.displayedAssets.find(asset => symbol === asset.symbol)) {
+        this.displayedAssets.push(assets[symbol]);
+      }
+    });
+  }
+
+  getTabs = () => {
+    return [
+      {
+        id: DISPLAYED,
+        name: 'Displayed',
+        onPress: () => this.setState({ activeTab: DISPLAYED }),
+      },
+      {
+        id: ALL_NON_ZERO,
+        name: 'All non-zero',
+        onPress: () => this.setState({ activeTab: ALL_NON_ZERO }),
+      },
+    ];
+  }
+
+  handleAssetToggle = (asset: Asset, added: Boolean) => {
+    if (!added) {
+      this.addTokenToWallet(asset);
+    } else {
+      this.hideTokenFromWallet(asset);
+    }
+  };
+
+  addTokenToWallet = (asset: Asset) => {
+    const { addAsset } = this.props;
+    addAsset(asset);
+  };
+
+  hideTokenFromWallet = (asset: Asset) => {
+    const {
+      hideAsset,
+    } = this.props;
+
+    if (asset.symbol === ETH || asset.symbol === PLR) {
+      this.showNotHiddenNotification(asset);
+      return;
+    }
+
+    Alert.alert(
+      'Are you sure?',
+      `This will hide ${asset.name} from your wallet`,
+      [
+        { text: 'Cancel', style: 'cancel' },
+        { text: 'Hide', onPress: () => hideAsset(asset) },
+      ],
+    );
+  };
+
+  showNotHiddenNotification = (asset) => {
+    Toast.show({
+      message: `${asset.name} is essential for Pillar wallet`,
+      type: 'info',
+      title: 'This asset cannot be switched off',
+    });
+  };
+
+  selectAllAssets = () => {
+    const { assets, addMultipleAssets } = this.props;
+    const allAssets = this.getAssetsList();
+    const assetsToAdd = allAssets.filter(asset => !assets[asset.symbol]);
+    addMultipleAssets(assetsToAdd);
+  }
+
+  handleSearchChange = (query: string) => {
+    const formattedQuery = !query ? '' : query.trim();
+
+    this.setState({
+      query: formattedQuery,
+    });
+
+    this.doAssetsSearch(formattedQuery);
+  };
+
+  doAssetsSearch = (query: string) => {
+    const { searchAssets, resetSearchAssetsResult } = this.props;
+    if (query.length < MIN_QUERY_LENGTH) {
+      resetSearchAssetsResult();
+      return;
+    }
+    searchAssets(query);
+  };
+
+  isInSearchMode = () => this.state.query.length >= MIN_QUERY_LENGTH && !!this.props.assetsSearchState;
```

Comment:
Assuming the same type, but prefer:

```suggestion
  isInSearchMode = () => this.state.query.length >== MIN_QUERY_LENGTH && !!this.props.assetsSearchState;
```

... for safety.

Reply:
There is no such operator like `>==` in JS. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators

Reply:
oh whoops! i saw `==`, this is what happens when i review code at 1am in the morning!